galette/webroot/installer.php

```diff
@@ -111,6 +114,18 @@ function initDbConstants($install)
     Analog::handler($galette_run_log);
 }
 
+if (!$install->isEndStep()
+    && ($install->postCheckDb() || $install->isDbCheckStep())
+) {
+    //if we have passed database configuration, define required constants
+    initDbConstants($install);
+
+    if ($install->postCheckDb()) {
+        //while before check db, connection is not checked
+        $zdb = new GaletteDb();
+    }
+}
+
```

Comment:
This change causes the following error on my side (wether I choose mysql or postgresql as database type on the database step) :

`PHP Fatal error:  Uncaught Error: Undefined constant "Galette\\Core\\TYPE_DB" in /var/www/galette/galette/lib/Galette/Core/Db.php:110\nStack trace:\n#0 /var/www/galette/galette/install/steps/db_checks.php(55): Galette\\Core\\Db->__construct()\n#1 /var/www/galette/galette/webroot/installer.php(384): include_once('...')\n#2 {main}\n  thrown in /var/www/galette/galette/lib/Galette/Core/Db.php on line 110`

I could only complete the install process reverting this change.
I don't know if it is the right thing to do...

Reply:
Indeed, that cannot work... We check if we're at database checks steps; but steps are defined after. Just reverting should not solve the issue; I do not remember right now why I've moved this part, but I guess I had an issue with the new step added.

I'll investigate

Reply:
Fixed in last commit

Reply:
> Fixed in last commit

Now an error is thrown on the database installation step :

`PHP Fatal error:  Uncaught ErrorException: Undefined variable $zdb in /var/www/galette/galette/install/steps/db_install.php:41\nStack trace:\n#0 /var/www/galette/galette/install/steps/db_install.php(41): {closure}()\n#1 /var/www/galette/webroot/installer.php(303): include_once('...')\n#2 {main}\n  thrown in /var/www/galette/galette/install/steps/db_install.php on line 41, referer: http://galette.localhost/installer.php`

Reply:
Argh... Indeed, I've tried to make things fast, and I just made things wrong :D

Reply:
It should be OK now

Reply:
Yes :+1: